src/c_api.cpp

```diff
@@ -1042,21 +1084,27 @@ int LGBM_DatasetCreateFromMats(int32_t nmat,
         offset += nrow[j];
         ++j;
       }
-
+      if(config.use_ctr > 0) {
+        sample_labels.push_back(static_cast<LightGBM::label_t>(get_label_fun(idx)));
+      }
       auto row = get_row_fun[j](static_cast<int>(idx - offset));
       for (size_t k = 0; k < row.size(); ++k) {
-        if (std::fabs(row[k]) > kZeroThreshold || std::isnan(row[k])) {
+        // if ctr is used, zero categorical values should not be ignored
+        if (std::fabs(row[k]) > kZeroThreshold || std::isnan(row[k]) || (config.use_ctr > 0 && is_categorical_feature[k])) {
```

Comment:
is CTR calculated by full dataset or sample set? sampled label is needed?

Reply:
It is calculated by full dataset. I will remove the sample label stuff.